pandas/core/generic.py

```diff
@@ -73,6 +87,13 @@
 from pandas.io.formats.printing import pprint_thing
 from pandas.tseries.frequencies import to_offset
 
+if TYPE_CHECKING:
+    from pandas import Series, DataFrame
+
+bool_t = bool
+FrameOrSeries = TypeVar("FrameOrSeries", bound="NDFrame")
+_T = TypeVar("_T")
```

Comment:
xref #28173

Reply:
> What is _T?

unconstrained type variable used in _protect_consolidate to preserve return type from callable.

_is_numeric_mixed_type, _is_mixed_type etc. call _protect_consolidate and returns a bool

_consolidate returns a DataFrame or Series if inplace=False

if a union was used for the return type of _protect_consolidate would require casts to allow return types of _consolidate, _is_numeric_mixed_type etc. to be annotated.

> Why do we need an alias for bool?

Need alias because NDFrame has def bool...

```
    def bool(self):
        """
        Return the bool of a single element PandasObject.
```